azure-log-analytics/microsoft_api_auth.py

```diff
@@ -28,236 +29,106 @@
+                                     connector_config['config_id'])
+            connector_config['config_id'] = config_id
+            return "Bearer {0}".format(connector_config.get('accessToken'))
         else:
-            client_id = connector_config.get('client_id')
-            section_header = 'Microsoft-API-Auth-{0}'.format(client_id)
-            time_key = ['expiresOn']
-            token_key = ['accessToken']
-            try:
-                config = self.handle_config(section_header)
-                ts_now = time()
-                expires = config.get(section_header, 'expiresOn')
-                if ts_now > float(expires):
-                    REFRESH_TOKEN_FLAG = True
-                    self.refresh_token = config.get(section_header, 'refresh_token')
-                    logger.info("Token expired at {0}".format(str(expires)))
-                    new_token = self.generate_token(REFRESH_TOKEN_FLAG)
-                    for key, val in new_token.items():
-                        if key in time_key:
-                            config.set(section_header, str(key), self.convert_ts_epoch(new_token.get(key)))
-                        if key in token_key:
-                            config.set(section_header, str(key), self.encode_token(new_token[key]).decode('utf-8'))
-
-                    with open(configfile, 'w') as fobj:
-                        config.write(fobj)
-                else:
-                    logger.info("Token is valid till {0}".format(str(expires)))
-
-                encoded_token = config.get(section_header, 'accessToken')
-                decoded_token = b64decode(encoded_token.encode('utf-8'))
-                token = "Bearer {0}".format(decoded_token.decode('utf-8'))
-                return token
-            except Exception as err:
-                logger.error("{0}".format(str(err)))
-                raise ConnectorError("{0}".format(str(err)))
-
-    def remove_config(self):
-        try:
-            section_header = 'Microsoft-API-Auth-{0}'.format(self.client_id)
-            self.handle_config(section_header, flag=True)
-        except Exception as err:
-            logger.error("{0}".format(str(err)))
-            raise ConnectorError("{0}".format(str(err)))
+            logger.info("Token is valid till {0}".format(expires))
+            return "Bearer {0}".format(connector_config.get('accessToken'))
+
+    def validate_log_token(self, connector_config, connector_info):
+        ts_now = time()
+        expires = connector_config['logExpiresOn']
+        expires_ts = self.convert_ts_epoch(expires)
+        if ts_now > float(expires_ts):
+            logger.info("Token expired at {0}".format(expires))
+            token_resp = self.generate_token(LOG_SCOPE)
+            connector_config['logAccessToken'] = token_resp['accessToken']
+            connector_config['logExpiresOn'] = token_resp['expiresOn']
+            connector_config['logRefresh_token'] = token_resp.get('refresh_token')
```

Comment:
Why this key is not in camelcase format? is there any requirement we need this key like this? if not then try to maintain all keys either camelcase or snake_case.